src/components/Viewport.ts

```diff
@@ -168,12 +170,25 @@ export default class Viewport {
     // Update position & nearestPanel
     state.position = pos;
     this.nearestPanel = this.findNearestPanel();
-
     const nearestPanel = this.nearestPanel;
     const originalNearestPosition = nearestPanel
       ? nearestPanel.getPosition()
       : 0;
 
+    // From 0(panel position) to 1(panel position + panel size)
+    // When it's on gap area value will be (val > 1 || val < 0)
+    if (nearestPanel) {
+      const hangerPosition = this.getHangerPosition();
+      const panelPosition = nearestPanel.getPosition();
+      const panelSize = nearestPanel.getSize();
+      const halfGap = options.gap / 2;
+
+      // As panel's range is from panel position - half gap ~ panel pos + panel size + half gap
+      state.panelMaintainRatio = (hangerPosition - panelPosition + halfGap) / (panelSize +  2 * halfGap);
```

Comment:
@WoodNeck How about to change from `hangerPosition` to `cameraPosition`?

Reply:
It's meaningful to use hanger position, as where camera position actually affected by panel is hanger position, not camera position.

Anyway, it should be related with hanger position.
If using camera position instead, it would be like
```js
state.panelMaintainRatio = (cameraPosition - (panelPosition - relativeHangerPosition - halfGap)) / (panelSize + 2 * halfGap);
```

Reply:
@WoodNeck I got it! Thank you!